### PR TITLE
Convert ResponseMap to Kotlin

### PR DESCRIPTION
**Pull request by JSunde**

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/model/submission/ResponseMap.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            responses[taskId]
+        )
+    }
+
+    /**
+     * Returns an Iterable over the task ids in this map.
+     */
+    fun taskIds(): Iterable<String> {
+        return responses.keys
+    }
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/model/submission/ResponseMap.kt:45]: Please convert to single line method

File: ground/src/main/java/com/google/android/ground/model/submission/ResponseMap.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    /**
+     * Returns the user response for the given task id, or empty if the user did not specify a
+     * response.
+     */
+    fun getResponse(taskId: String): Optional<Response> {
+        return Optional.ofNullable(
+            responses[taskId]
+        )
+    }
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/model/submission/ResponseMap.kt:38]: Please convert to single line method

File: ground/src/main/java/com/google/android/ground/persistence/local/room/converter/ResponseMapConverter.kt [hunk 3]
```diff
@@ -27,26 +29,23 @@ import timber.log.Timber
...
+                Timber.e(e, "Error building JSON")
             }
-        }.toString()
+        }
+    }.toString()
 
     @JvmStatic
     fun fromString(job: Job, jsonString: String?): ResponseMap {
-        val map = ResponseMap.builder()
+        val map = ImmutableMap.builder<String, Response>()
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/persistence/local/room/converter/ResponseMapConverter.kt:46]: Please move this after null check

File: ground/src/main/java/com/google/android/ground/persistence/remote/firestore/schema/SubmissionConverter.kt [hunk 4]
```diff
@@ -58,9 +59,9 @@ internal object SubmissionConverter {
     private fun toResponseMap(
         submissionId: String, job: Job, docResponses: Map<String, Any>?
     ): ResponseMap {
-        val responses = ResponseMap.builder()
+        val responses = ImmutableMap.builder<String, Response>()
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/persistence/remote/firestore/schema/SubmissionConverter.kt:62]: Please move this after null check

**Review by shobhitagarwal1612 — CHANGES_REQUESTED** [review]

File: ground/src/main/java/com/google/android/ground/persistence/remote/firestore/schema/SubmissionConverter.kt (hunk 4 above)
Comment (JSunde) [reply ground/src/main/java/com/google/android/ground/persistence/remote/firestore/schema/SubmissionConverter.kt]: Done!

File: ground/src/main/java/com/google/android/ground/model/submission/ResponseMap.kt (hunk 1 above)
Comment (JSunde) [reply ground/src/main/java/com/google/android/ground/model/submission/ResponseMap.kt:45]: Done.

File: ground/src/main/java/com/google/android/ground/model/submission/ResponseMap.kt (hunk 2 above)
Comment (JSunde) [reply ground/src/main/java/com/google/android/ground/model/submission/ResponseMap.kt:38]: Done!

**Review by shobhitagarwal1612 — APPROVED** [review]
LGTM, thanks!

File: ground/src/main/java/com/google/android/ground/persistence/local/room/converter/ResponseMapConverter.kt (hunk 3 above)
Comment (JSunde) [reply ground/src/main/java/com/google/android/ground/persistence/local/room/converter/ResponseMapConverter.kt]: Done!

**Pull request merged**